framework/src/org/checkerframework/framework/util/typeinference/DefaultTypeArgumentInference.java

```diff
@@ -135,18 +139,41 @@ public DefaultTypeArgumentInference(AnnotatedTypeFactory typeFactory) {
         }
 
         final Set<TypeVariable> targets = TypeArgInferenceUtil.methodTypeToTargets(methodType);
-        final Map<TypeVariable, AnnotatedTypeMirror> inferredArgs =
-                infer(typeFactory, argTypes, assignedTo, methodElem, methodType, targets, true);
-
-        if (showInferenceSteps) {
-            checker.message(Kind.NOTE, "  after infer: %s\n", inferredArgs);
-        }
-
-        handleNullTypeArguments(
-                typeFactory, methodElem, methodType, argTypes, assignedTo, targets, inferredArgs);
-
-        if (showInferenceSteps) {
-            checker.message(Kind.NOTE, "  after handleNull: %s\n", inferredArgs);
+        Map<TypeVariable, AnnotatedTypeMirror> inferredArgs;
+        if (assignedTo == null && TreeUtils.getAssignmentContext(pathToExpression) != null) {
+            inferredArgs = new LinkedHashMap<>();
+        } else {
+            try {
```

Comment:
Is it worth putting this whole else branch into a new helper method and slap on some comment why we're doing this?